service/health/searcher/service.go

```diff
@@ -10,37 +10,42 @@ import (
 
 // Config represents the configuration used to create a new service object.
 type Config struct {
-	// Dependencies.
 	G8sClient versioned.Interface
 	Logger    micrologger.Logger
 
 	Provider string
 }
 
+type Service struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+
+	provider string
+}
+
 // New creates a new configured service object.
 func New(config Config) (*Service, error) {
-	// Dependencies.
 	if config.G8sClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "g8sclient must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
 	}
 	if config.Logger == nil {
-		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
+
 	if config.Provider == "" {
-		return nil, microerror.Maskf(invalidConfigError, "provider must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.Provider must not be empty", config)
 	}
 
 	newService := &Service{
```

Comment:
IMO this should have the same name as in the method receiver:

```suggestion
	s := &Service{
```